### DISCUSSION TITLE
Несколько вопросов

### DISCUSSION THREAD

**Discussion by Coockie** [Try/OpenGothic#265] [Q&A]
Правильно ли я понимаю, что вы Try используете собственный, написанный вами с нуля движок? И он использует АПИ Вулкан и Direct3D 12?
И ещё пара вопросов: почему такие большие файлы? 200 с чем-то Мб екзешник и несколько десятков Мб Tempest? Это из-за отладочных символов? Или там ещё какие-то сторонние файлы вкомпилены? Так же ещё сборка 64b ночников компилятором MSVC2019-ым выдаёт файлы значительно меньших размеров и игра при этом не запускается, это нормально?

**Comment by Try** [comment]
Привет, @Coockie!

Да, движок свой на основе Vulkan(основной рендер), DX12 и Metal.
200 мб - это mingw сборка, там отладочная инфа зашита в *.exe. В MSVC сборках такого нет - им было уделено меньше внимания.

>  и игра при этом не запускается, это нормально?

А как именно не запускается? У меня из *.zip не запустилась(видимо сам winrar не дает доступа к файловой системе), а распакованный вариант - работает.

ЗЫ:
If, possible please stick to English, so everyone can read/engage in conversation. Thanks!

**Reply by Coockie** [reply]
Here is screenshot of error. I am using GOG russian version of Gothic 2 on Windows 10 and Windows 11 on two different computers. Logs are empty.
![OpenGothic_dxcompiler](https://user-images.githubusercontent.com/6870054/166869135-4349adf0-e50f-463d-921a-419bd6ac674a.jpg)

**Reply by Coockie** [reply]
Скачал из интернета недостающий файл dxcompiler.dll и игра запустилась. Правда ФПС по сравнению с отладочной версией нифига не вырос, а даже наоборот не много уменьшился: было на debug сборке 27-28 стало 25-28 при разрешении 1680x1050 в самом начале сразу после окончания разговора с Ксардасом. Система: CPU AMD Athlon 5350 GPU AMD Radeon RX 552 OS Win 10 21H2
Ещё обнаружил пару проблем: 
1. Не отображается буква "я" в текстах игры. Например в самом начале в башне нет буквы "я" у "статуя белиара" вместо этого пишется "стату белиара" также нет этой буквы в "лечебная трава" и книгах на вершине башни. Вот скриншот из оригинальной игры где эта буква есть. 
![Gothic_bukva_ya](https://user-images.githubusercontent.com/6870054/166876668-1748939d-1aa8-4383-a07b-a6c39e31cac5.jpg)
2. Первый ролик, т.е. сразу после нажатия на "Новая игра" чуть-чуть подёргивается, а во втором бывают вообще заикания звука начинаются на пятой секунде видео хотя видеопоток идёт нормально, но чаще всего не большие заикания всё таки проскакивают. Это в разрешении 1680x1050, в оригинальной версии при разрешении 800x600 всё видео проигрывается хорошо.
3. Ещё я не могу сделать скриншот: нажимаю PrntScrn и вставляю в mspaint, а там всё белое вместо скриншота игры.

**Comment by Try** [comment]
> было на debug сборке 27-28 стало 25-28 

You actually comparing 2 different kinds of RelDebug builds, so performance should be similar.
I'm not sure how good 'RX 552', but you may try to disable options->extended->cloud shadows, and see if it helps with gpu performance.

> Не отображается буква "я" в текстах игры. 

Hm, I use to had same on Akella CD version back in a days, with vanilla game, - will check.

> Это в разрешении 1680x1050

Video is done fully on CPU due to implementation complexity(*.bink is proprietary video format) - should do proper steaming eventually.
PS: video is actually in 320x200

> не могу сделать скриншот: 

I guess and driver forces exclusive mode(I have same issue on intell UHD driver). If it's really necessary,  you can enable 'marvin' mode and switch to windowed rendering by pressing F3.

**Reply by Coockie** [reply]
>You actually comparing 2 different kinds of RelDebug builds, so performance should be similar.

So why is then one archive 5MiB and another is ~90MiB?

>I'm not sure how good 'RX 552'

It's slightly faster than Nvidia Geforce GTX 1030 and probably between Geforce GTX 750 and GTX 750 Ti. CPU is also potato. It have 20% less performance in single core than a mobile Core 2 Celeron T2100(2011-2013 years) that working at 2100MHz.

>but you may try to disable options->extended->cloud shadows, and see if it helps with gpu performance.

It has been already disabled if i enable this option then performance in that scene slighly drops.

Is it normal that when i watch in taskmanager the game CPU and GPU utilization are only 50%? Also MSI Afterburner overlay doesn't work so i can't watch CPU and GPU utilization in game.

**Comment by Try** [comment]
> So why is then one archive 5MiB and another is ~90MiB?

I MinGW debug information in embedded into *.exe, in MSVC - separated *.pdb. And  *.pdb does not land into archive.

> Is it normal that when i watch in taskmanager the game CPU and GPU utilization are only 50%?

It seems on your machine game is CPU bound. CPU usage is quite a topic: big view range means big amount of draw-calls and there is no great way to ask Vulkan to `vkCmdDrawAll`. Also, if you interested, there is #188 with more performance-related stuff.

> CPU ... utilization are only 50%?

Generally - yes. It's hard to get 100% CPU capacity in a real application. And GPU idling, while waiting for new commands from CPU.
